ui/app/src/components/Header.tsx

```diff
@@ -48,45 +51,55 @@ function ProjectMenu(): JSX.Element {
     return <CircularProgress size="1rem" />;
   }
 
-  if (data !== undefined) {
-    return (
-      <>
-        <Button
-          aria-label="List of the available projects"
-          aria-controls="menu-project-list-appbar"
-          aria-haspopup="true"
-          color="inherit"
-          endIcon={<ChevronDown />}
-          onClick={handleMenu}
-        >
-          Projects
-        </Button>
-        <Menu
-          id="menu-project-list-appbar"
-          anchorEl={anchorEl}
-          anchorOrigin={{
-            vertical: 'bottom',
-            horizontal: 'left',
-          }}
-          keepMounted
-          open={anchorEl !== null}
-          onClose={handleCloseMenu}
-        >
-          {data.map((value, index) => {
-            return (
-              // TODO when routing is in place, use the button to redirect to the project page
-              <MenuItem key={index} onClick={handleCloseMenu}>
-                {value.metadata.name}
-              </MenuItem>
-            );
-          })}
-        </Menu>
-      </>
-    );
+  if (data === undefined) {
+    return <></>;
   }
-  // In case the loading is finished and there is an error, it will be handled by the snackbar.
-  // That's why we return an empty bracket
-  return <></>;
+
+  return (
+    <>
+      <Button
+        aria-label="List of the available projects"
+        aria-controls="menu-project-list-appbar"
+        aria-haspopup="true"
+        color="inherit"
+        endIcon={<ChevronDown />}
+        onClick={handleMenu}
+      >
+        Projects
+      </Button>
+      <Menu
+        id="menu-project-list-appbar"
+        anchorEl={anchorEl}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'left',
+        }}
+        keepMounted
+        open={anchorEl !== null}
+        onClose={handleCloseMenu}
+        PaperProps={{
+          style: {
+            maxHeight: ITEM_HEIGHT * 4.5,
+            width: '20ch',
```

Comment:
Maybe an inline comment here could be useful on why using 'ch' here is needed? Not saying it isn't justified, just isn't something you see too often

Reply:
I put `10em` instead